src/test/java/org/broadinstitute/hellbender/utils/haplotype/HaplotypeBAMWriterUnitTest.java

```diff
@@ -340,4 +338,14 @@ private int getReadCounts(final Path result) throws IOException {
         return count;
     }
 
+    private List<SAMReadGroupRecord> getReadGroups(final Path result) {
+        IOUtil.assertFileIsReadable(result);
+
+        try (final SamReader in = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT).open(result)) {
+            return in.getFileHeader().getReadGroups();
+        } catch (IOException e) {
+            throw new UserException("Unable to open " + result.toString());
+        }
+    }
```

Comment:
This method appears to currently be unused, but you can use it to verify that the id matches the one requested in the test above (see my comment elsewhere).